### PR TITLE
Add chpldoc tests of --author

### PR DESCRIPTION
**Pull request by Rapiz1**
Add chpldoc tests of the flag `--author`

resolves https://github.com/chapel-lang/chapel/issues/6029

@lydia-duncan Please review this PR : )

### CODE REVIEW

File: test/chpldoc/compflags/author/CATFILES [hunk 1]
```diff
@@ -0,0 +1 @@
+docs/modules/author.doc.html
```
Comment (lydia-duncan) [inline test/chpldoc/compflags/author/CATFILES]: We try to reserve all caps files for general operations that should be performed on any tests that might be added to the directory.  The CLEANFILES you've added here fits that category, but I think the other three are more test-specific, so should be applied to just the test itself

**Comment by lydia-duncan** [comment]
Thanks for taking this on :)

**Comment by Rapiz1** [comment]
I think I just find a bug when playing around
image
When --author is not specified, I think it should be `Copyright 2015` withou `,.`
- - -
Also, for `mason doc` I got this
image
This should match the `authors` field in `Mason.toml`, shouldn't it?
- - -
I'd like to fix this and add tests. Can you give me some guidance about where to start?

**Comment by Rapiz1** [comment]
I just figured it out how to fix it and opened a new PR since it's another bug and there may need to be discussion about how `mason doc` should deal with `authors` in `Mason.yaml`, while this PR is about testing `--author`.
And I have made the recommended changes using file-specific helper files. But I'm still thinking about what other cases can be.
What about testing special characters like `<something>` being escaped in case of someone who wants to include his email address like `rapiz<contact@rapiz.me>` but breaking HTML? However, I think it's more like the responsibility for `sphinx` to test that.
That's all I can come up with for now. Do you have any other ideas?

**Comment by lydia-duncan** [comment]
> When --author is not specified, I think it should be Copyright 2015 without ,.

I agree :)

> This should match the authors field in Mason.toml, shouldn't it?

Yeah, that sounds right.  I'll let conversation continue on the Mason specific issue.

> What about testing special characters like `<something>` being escaped in case of someone who wants to include his email address like `rapiz<contact@rapiz.me>` but breaking HTML? However, I think it's more like the responsibility for sphinx to test that.

That's an interesting idea.  I'd be curious what happened in that case for sure.  I was thinking more like what happens when there's a space in the name, what happens if something that could be recognized as a bash environment variable was used, may some other stuff.  Looking at where author is implemented could give you other ideas - you found some parts already, but you might look at compiler/passes/docs.cpp as well

**Comment by lydia-duncan** [comment]
(The test ran just fine for me locally, so only waiting on if additional tests should be added)

**Comment by Rapiz1** [comment]
I am glad to hear that! I will add more tests at a later time.

**Comment by Rapiz1** [comment]
More tests are added.
>  what happens if something that could be recognized as a bash environment variable was used

Unfortunately, I have added such a test and it failed. I think it's an issue about `start_test` instead of `chpldoc` itself since it works well when invoking from terminal. I will try to figure it out. ( Still, maybe another PR ) It seems that we can get a lot from testing `--author` : )

**Comment by Rapiz1** [comment]
@lydia-duncan I'm stuck with `sub_test.py` replacing everything in `withEnv.doc.chpldocopts`(as you suggested, adding a test passing `--author '$PATH'`) that starts with a `$`. I opened a [PR](https://github.com/chapel-lang/chapel/pull/15170) to solve this problem but I'm not sure that's appropriate.

**Comment by lydia-duncan** [comment]
Sorry, I didn't get back to this today, will try to tomorrow

**Comment by lydia-duncan** [comment]
These new tests look good, though I will want to check the generated documentation output for the html one to make sure it looks okay.  Taking a look at your `$` PR now

**Comment by Rapiz1** [comment]
I have read your comment https://github.com/chapel-lang/chapel/pull/15170#issuecomment-600150997 and I suppose I should delete the test `withEnv`, right?

**Comment by lydia-duncan** [comment]
Yeah, I think that's the right call.  Thanks for looking into it, I feel better knowing that it worked when run from the terminal itself

**Comment by Rapiz1** [comment]
@lydia-duncan I have made the change. Sorry for the delay. I had a false memory that I had made the change days before ...

**Pull request merged**